addons/test_portal/models.py

```diff
@@ -0,0 +1,17 @@
+from odoo import models
+
+
+class NotPortal(models.Model):
+    _name = _description = 'test_portal.not_portal'
+
+    def _compute_access_url(self):
+        super()._compute_access_url()
+        self.access_url = '/my/thing2'
+
+class Portal(models.Model):
```

Comment:
Could you maybe add some basic fields (email, name, partner_id, ...) ? That way I will re-use it soon to test some notification flows with portal (as test_mail does not check portal). If you don't want because "minimal diff", Rochers Suchards will be required.